pkg/cpu/cpu_linux.go

```diff
@@ -64,6 +65,10 @@ func processorsGet(ctx *context.Context) []*Processor {
 			continue
 		}
 
+		onlineFilePath := filepath.Join(paths.SysDevicesSystemCPU, fmt.Sprintf("cpu%d", lpID), onlineFile)
+		if util.SafeIntFromFile(ctx, onlineFilePath) == 0 {
```

Comment:
this will return -1 if the file does not exist... perhaps it's worth changing this to:

```go
if util.SafeIntFromFile(ctx, onlineFilePath) != 1 {
```

in order to account for when there are problems locating the `/online` file?

Reply:
I agree, will make the required changes.

Thanks!

Reply:
It was observed that this change didn't fit well for the ARM architecture, as the `online` file does not exist under the  cpuX directory. This lead to the mismatch in the expected vs observed tests results causing a breaking change. 

The `online` file exists for the amd64/ppc64le architecture where, the corresponding files are in the topology directory may/may not exist based on the bit set in the `online` file.  I have made changes to account only if the cpu is offline, before the files under the topology directory are checked. This doesn't seem to affect the `arm64` architecture as the `-1` returned from the caller doesn't affect the rest of the code execution. 

Thanks!